presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -408,15 +389,49 @@ private PlanBuilder aggregate(PlanBuilder subPlan, QuerySpecification node)
             groupingSetsSymbolsBuilder.add(groupingColumns.build());
         }
 
+        // 2.b. Add a groupIdNode and groupIdSymbol if there are multiple grouping sets
         List<List<Symbol>> groupingSetsSymbols = groupingSetsSymbolsBuilder.build();
-        // 2.c. Add a groupIdNode and groupIdSymbol if there are multiple grouping sets
         if (groupingSets.size() > 1) {
+            Set<Symbol> groupIdInputSymbols = new HashSet<>();
+            distinctGroupingColumns.stream()
+                    .map(subPlan::translate)
+                    .forEach(groupIdInputSymbols::add);
+
+            ImmutableMap.Builder<Symbol, Symbol> passthroughSymbolMap = ImmutableMap.builder();
```

Comment:
These names are too verbose: passthroughSymbolMap, passthroughAggregationArgument, groupIdInputSymbols, passthroughSymbol. You can just name them "mappings", "argument", "inputs", "output" without loss of clarity.